ui/selectors/selectors.js

```diff
@@ -1334,13 +1335,59 @@ export function getAllAccountsOnNetworkAreEmpty(state) {
   return hasNoNativeFundsOnAnyAccounts && hasNoTokens;
 }
 
-export function getShouldShowSeedPhraseReminder(state) {
-  const { tokens, seedPhraseBackedUp, dismissSeedBackUpReminder } =
-    state.metamask;
+export function getIsAnyAccountWithBalanceOnNonTestNetwork(state) {
+  const { cachedBalances } = state.metamask;
+
+  const zeroBalanceCheck = (currentValue) =>
+    currentValue === '0x0' || currentValue === '0x00';
+
+  const balancesOnNonTestNetworks = Object.entries(cachedBalances)
+    .filter(
+      (networkAndAccountBalances) =>
+        TEST_CHAINS.includes(networkAndAccountBalances[0]) === false &&
+        networkAndAccountBalances,
+    )
+    .map((accountAndBalance) => Object.values(accountAndBalance[1]))
+    .map((balance) => balance.every(zeroBalanceCheck));
+
+  return balancesOnNonTestNetworks.includes(false);
+}
+
+export async function getIsAnyTokenWithNonZeroBalance(state) {
+  const { tokens } = state.metamask;
+  const selectedAddress = getSelectedAddress(state);
+
+  if (tokens.length > 0) {
+    for (let i = 0; i < tokens.length; i++) {
+      const res = await getERC20Balance(tokens[i], selectedAddress);
+
+      if (res !== '0x0') {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+export async function getShouldShowSeedPhraseReminder(state) {
+  const { seedPhraseBackedUp, dismissSeedBackUpReminder } = state.metamask;
```

Comment:
A large amount of computation could be shortcut in most cases by adding this here:
`if (seedPhraseBackedUp || dismissSeedBackUpReminder) return false;`

Reply:
A little later I did some performance tests, and this function `getShouldShowSeedPhraseReminder()` runs quite frequently.  It basically runs at least two times whenever you click any button, anywhere in the UI, and it runs for 25-35 ms.